src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/ExactTypeBindingProvider.cs

```diff
@@ -96,7 +102,14 @@ private class ExactBinding : BindingBase<TAttribute>
 
                 IValueProvider vp = new ConstantValueProvider(obj, typeof(TUserType), invokeString);
                 return vp;
-            }            
+            }
+
+            protected override Task<IValueProvider> BuildAsync(TAttribute attrResolved, BindingContext context)
+            {
+                _preBindHook?.Invoke(attrResolved, context);
```

Comment:
I've only plumbed this through ExactBinding, not the other binding types. If we agree on expanding the hooks like this (I think we need to anyways) we need to do it a cross the board.

Loose delegates like this make it cumbersome for binding implementations to hook in. As the number of hook points increases, I think it'll get worse. They really need to be specifying a custom type defining the contract, allowing them to override only the bits they need.